RecoBTag/DeepFlavour/interface/sorting_modules.h

```diff
@@ -0,0 +1,187 @@
+ */
+
+template<class T>
+bool comparePt(T a, T b){
+    if(!a)return true;
+    if(!b)return false;
+    return a->pt()<b->pt();
+}
+
+
+template <class T>
+class sortingClass{
+public:
+
+    sortingClass():sortValA(0),sortValB(0),sortValC(0),t_(0){}
+
+    sortingClass(const T& t, float sortA, float sortB=0, float sortC=0){
+        t_=t;
+        sortValA=sortA;
+        sortValB=sortB;
+        sortValC=sortC;
+    }
+    sortingClass(const sortingClass&rhs):
+    sortValA(rhs.sortValA),sortValB(rhs.sortValB),sortValC(rhs.sortValC),t_(rhs.t_)
+    {	}
+    
+    sortingClass& operator=(const sortingClass&rhs){
+    	sortValA=(rhs.sortValA);
+    	sortValB=(rhs.sortValB);
+    	sortValC=(rhs.sortValC);
+    	t_=(rhs.t_);
+    	return *this;
+    }
+
+    const T& get()const{return t_;}
+
+    //hierarchical sort
+    static bool compareByABC(sortingClass a, sortingClass b){
+        if(std::isnormal(a.sortValA) && std::isnormal(b.sortValA) && a.sortValA!=b.sortValA){
+            return CompareA(a,b);
+        }
+        else if(!std::isnormal(a.sortValA) && std::isnormal(b.sortValA)){
+            return true;
+        }
+        else if(std::isnormal(a.sortValA) && !std::isnormal(b.sortValA)){
+            return false;
+        }
+        else{
+            if(std::isnormal(a.sortValB) && std::isnormal(b.sortValB) && a.sortValB!=b.sortValB){
+                return CompareB(a,b);
+            }
+            else if(!std::isnormal(a.sortValB) && std::isnormal(b.sortValB)){
+                return true;
+            }
+            else if(std::isnormal(a.sortValB) && !std::isnormal(b.sortValB)){
+                return false;
+            }
+            else{
+                if(std::isnormal(a.sortValC) && std::isnormal(b.sortValC) && a.sortValC!=b.sortValC){
+                    return CompareC(a,b);
+                }
+                else if(!std::isnormal(a.sortValC) && std::isnormal(b.sortValC)){
+                    return true;
+                }
+                else if(std::isnormal(a.sortValC) && !std::isnormal(b.sortValC)){
+                    return false;
+                }
+                else{
+                    return true;
+                }
+            }
+        }
+        return true; //never reached
+    }
+
+    static bool compareByABCInv(sortingClass a, sortingClass b){
+        return !compareByABC(a,b);
+    }
+
+ private:
+    float sortValA,sortValB,sortValC;
+    
+    static bool CompareA(sortingClass a, sortingClass b){
+        return a.sortValA<b.sortValA;
+    }
+    static bool CompareB(sortingClass a, sortingClass b){
+        return a.sortValB<b.sortValB;
+    }
+    static bool CompareC(sortingClass a, sortingClass b){
+        return a.sortValC<b.sortValC;
+    }
+    T t_;
+};
+
+
+
+std::vector<std::size_t> invertSortingVector(const std::vector<sortingClass<std::size_t> > & in);
+
+
+template<class T>
+bool compareDxy(T b, T a){
```

Comment:
where is this used?

Reply:
checked with @jkieseler and not needed anymore, will remove

Reply:
removed with https://github.com/cms-sw/cmssw/pull/19893/commits/6ad48f2a3527cf49f3a753cf57bfe30380048472